### ISSUE TITLE
Cop idea: compound hash initialization

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#12956]
A pattern I encounter often is where developers start with the `{}` Hash literal and proceed to populate each key one by one. Initializing the hash keys all at once is more readable and almost certainly more efficient.

```ruby
# bad
params = {}
params[:dog]  = 'bark'
params[:cat]  = 'meow'
params[:bird] = 'tweet'

# good
params = { dog: 'bark', cat: 'meow', bird: 'tweet' }
```

**Comment by koic** [comment]
Can you first open a proposal in the style guide to discuss the merits and demerits of this style?
https://github.com/rubocop/ruby-style-guide
